src/legacy-routes.js

```diff
@@ -14,36 +19,108 @@ const invalidTx = 'Invalid transaction id!'
 const arraySum = (numbers) => numbers.reduce((acc, val) => acc.plus(Big(val)), Big(0))
 
 /**
- * Helper function that takes movements for various addresses and a set of addresses we are
- * interested in. The sum of movements for addresses we are interested in is returned.
+ * Database stores all hashes with prefix of '\x'. To hide inner database structure and
+ * deal with just the results in common format, these functions wrap and unwrap the hashes.
 */
-const txAddressCoins = (addresses, amounts, addressSet) => arraySum(zip(addresses, amounts)
-  .filter((pair) => addressSet.has(pair[0]))
-  .map((pair) => nth(pair, 1)))
+const wrapHashPrefix = (hash: string): string => `\\x${hash}`
+const unwrapHashPrefix = (hash: string): string => hash.substr(2)
 
-const combinedBalance = (transactions, addresses) => {
-  const addressSet = new Set(addresses)
-  const totalIn = transactions.reduce((acc, tx) =>
-    acc.plus(txAddressCoins(tx.outputs_address, tx.outputs_amount, addressSet)), Big(0))
-  const totalOut = transactions.reduce((acc, tx) =>
-    acc.plus(txAddressCoins(tx.inputs_address, tx.inputs_amount, addressSet)), Big(0))
-  return totalIn.sub(totalOut)
+/**
+ * Initializes tx entry in the caTxList format into which inputs can be added and summed.
+ * @param {Tx} tx Transaction from database
+ */
+const initializeTxEntry = (tx: Tx) : TxEntry => ({
+  ctbId: unwrapHashPrefix(tx.hash),
+  ctbTimeIssued: moment(tx.time).unix(),
+  ctbInputs: [],
+  ctbOutputs: [],
+  ctbInputSum: { getCoin: Big(0) },
+  ctbOutputSum: { getCoin: Big(0) },
+})
+
+/**
+ * Initializes tx input/output entry
+ * @param {Movement} movement Tx input or output
+ */
+const initializeMovementEntry = (movement: Movement)
+: MovementEntry => [movement.address, { getCoin: movement.value }]
```

Comment:
MovementEntry sounds like it should be a Tuple[string, coinObject] rather than Array[Any]

https://www.typescriptlang.org/docs/handbook/basic-types.html#tuple